pkg/channel/consolidated/reconciler/controller/kafkachannel.go

```diff
@@ -225,34 +230,30 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, kc *v1beta1.KafkaChannel
 	// Reconcile the k8s service representing the actual Channel. It points to the Dispatcher service via ExternalName
 	svc, err := r.reconcileChannelService(ctx, dispatcherNamespace, kc)
 	if err != nil {
-
 		return err
 	}
 	kc.Status.MarkChannelServiceTrue()
 	kc.Status.SetAddress(&apis.URL{
 		Scheme: "http",
 		Host:   network.GetServiceHostname(svc.Name, svc.Namespace),
 	})
-	err = r.setupSubscriptionStatusWatcher(ctx, kc)
+	reconciled, err := r.reconcileSubscribers(ctx, kc)
 	if err != nil {
-		logger.Errorw("error setting up some subscription status watchers", zap.Error(err))
+		return fmt.Errorf("error reconciling subscribers %v", err)
 	}
-	// close the connection
-	err = kafkaClusterAdmin.Close()
-	if err != nil {
-		logger.Errorw("Error closing the connection", zap.Error(err))
-		return err
+	if !reconciled {
```

Comment:
why wouldn't this be considered a new error type and then we can warn on that?

Reply:
can you explain more? as in consider the case where not all subscribers were reconciled an error and check the error returned by `reconcileSubscribers` instead of a bool and error?

Reply:
right.  We're mixing the return value across multiple variables at this point.  For example, the fact we sometimes return `reconciled` and othertimes elect to return a constant (`false`) indicates to me, that something is amiss with the logic as it currently stands.

let's keep the single return value (of type `error`) and have the control flow dependant on the actual error.